distributed/worker.py

```diff
@@ -3468,75 +3480,126 @@ async def execute(self, key: str, *, stimulus_id: str) -> None:
                         self.scheduler_delay,
                     )
             finally:
-                self.active_keys.discard(ts.key)
-
-            key = ts.key
-            # key *must* be still in tasks. Releasing it directly is forbidden
-            # without going through cancelled
-            ts = self.tasks.get(key)  # type: ignore
-            assert ts, self.story(key)
-            ts.done = True
-            result["key"] = ts.key
-            value = result.pop("result", None)
-            ts.startstops.append(
-                {"action": "compute", "start": result["start"], "stop": result["stop"]}
-            )
-            self.threads[ts.key] = result["thread"]
-            recommendations: Recs = {}
+                self.active_keys.discard(key)
+
+            self.threads[key] = result["thread"]
+
             if result["op"] == "task-finished":
-                ts.nbytes = result["nbytes"]
-                ts.type = result["type"]
-                recommendations[ts] = ("memory", value)
                 if self.digests is not None:
                     self.digests["task-duration"].add(result["stop"] - result["start"])
```

Comment:
Same here. Do we want to move this to the result/event handler?

Reply:
self.digests is populated with a wealth of information that should remain alien to the state machine, so I think it should remain in Worker?

- latency
- transfer-bandwidth
- get-data-send-duration
- disk-load-duration
- profile-duration

Reply:
We can also nuke the digests if they're slowing us down.  They're not commonly used today.